prog/vnet/rekey_nic.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+class Prog::Vnet::RekeyNic < Prog::Base
+  subject_is :nic
+
+  def start
+    nic.dst_ipsec_tunnels.each do |tunnel|
+      args = frame["payload"][tunnel.src_nic.id]
+      create_state(tunnel, args)
+    end
+
+    hop :wait_outbound_trigger
+  end
+
+  def wait_outbound_trigger
+    if SemSnap.new(nic.id).set?("trigger_outbound_update")
+      SemSnap.new(nic.id).decr("trigger_outbound_update")
+      hop :setup_outbound
+    end
+    nap 1
+  end
+
+  def setup_outbound
+    nic.src_ipsec_tunnels.each do |tunnel|
+      args = frame["payload"][tunnel.src_nic.id]
+      create_state(tunnel, args)
+      policy_update(tunnel, "out")
+    end
+
+    hop :wait_state_drop_trigger
+  end
+
+  def wait_state_drop_trigger
+    if SemSnap.new(nic.id).set?("old_state_drop")
+      SemSnap.new(nic.id).decr("old_state_drop")
+      hop :old_state_drop
+    end
+
+    nap 1
+  end
+
+  def old_state_drop
+    new_spis = frame["payload"].values.flat_map { |p| [p["spi4"], p["spi6"]] }.flatten
```

Comment:
```suggestion
    new_spis = frame["payload"].values.flat_map { _1.fetch_values("spi4", "spi6")  }.flatten
```

It's longer for the symbol name but there's less subexpression going on, maybe it works as intended? Depends on some of your nil behavior goals.

Reply:
new pr doesn't have this https://github.com/ubicloud/ubicloud/pull/441